src/time.cr

```diff
@@ -503,6 +503,36 @@ struct Time
     utc(seconds: seconds, nanoseconds: nanoseconds.to_i)
   end
 
+  # Create a new `Time` instance that preserves the same wall clock reading than
+  # *time*  but in a different *location*.
+  #
+  # ```
+  # time = Time.now(Time::Location.new(-3600))
+  # time2 = Time.from_wall_clock(time, Time::Location.new(14400))
+  # time2.to_s("%Y-%m-%d %H:%M:%S") == time1.to_s("%Y-%m-%d %H:%M:%S") # => true
+  # ```
+  #
+  # Both instances refer to a different instant in time and they are apart from
+  # each other by the difference in zone offsets:
+  #
+  # ```
+  # time2 - time # => (time.offset - time2.offset).seconds
+  # ```
+  #
+  # The result of this method is equivalent to going through the calendrical
+  # constructor like this:
+  #
+  # ```
+  # time = Time.now
+  # time2 = Time.new(time.year, time.month, time.day, time.hour, time.minute, time.second, nanosecond: time.nanosecond, location: location)
+  # ```
```

Comment:
It might be helpful to talk about how this differs from `Time#in`. Right now it is difficult to tell which one to use and how they are different